pypsa/statistics.py

```diff
@@ -235,31 +262,41 @@ def aggregate_components(
         if n.df(c).empty:
             continue
 
-        if callable(groupby):
-            grouping = groupby(n, c, nice_names=nice_names)
-        elif isinstance(groupby, list):
-            grouping = [n.df(c)[key] for key in groupby]
-        elif isinstance(groupby, str):
-            grouping = n.df(c)[groupby]
-        elif isinstance(groupby, dict):
-            grouping = None
-            kwargs = groupby
-        elif groupby is not False:
-            ValueError(
-                f"Argument `groupby` must be a function, list, string, False or dict, got {type(groupby)}"
-            )
+        groupby_kwargs = to_groupby_kwargs(groupby, n, c, nice_names)
 
         df = func(n, c)
-        if isinstance(n.snapshots, pd.MultiIndex) and not isinstance(df, pd.DataFrame):
-            # for static values we have to iterate over periods and concat
-            per_period = {}
-            for p in n.investment_periods:
-                per_period[p] = df[n.get_active_assets(c, p).loc[df.index]]
-            df = pd.concat(per_period, axis=1)
+
+        if isinstance(n.snapshots, pd.MultiIndex):
```

Comment:
maybe this functionality is also useful for running myopic code where you don't have a multi-index snapshot but still the information about build year and lifetime